### ISSUE TITLE
Issues on attendance tracking feature

### ISSUE THREAD

**Issue by radinamatic** [learningequality/kolibri#14396]
<img height="20px" src="https://i.imgur.com/c7hUeb5.jpeg">

❌ **This issue is not open for contribution. Visit <a href="https://learningequality.org/contributing-to-our-open-code-base/" target="_blank">Contributing guidelines</a>** to learn about the contributing process and how to find suitable issues.

<img height="20px" src="https://i.imgur.com/c7hUeb5.jpeg">

**Target branch:** release-0.19.x
Tested the asset from #14382 on Windows 10, Ubuntu 22, and a tablet with Android 13.

## Observed behavior
1. When first I noticed that the arrow buttons for pagination on *Mark attendance* and *Edit attendance* pages were not keyboard navigable, it looked like it was only because of low number of learners, but even when I enrolled more (from 16 to 46), the list was still not paginated. Not sure where the threshold is for the number of learners in the class to be moved to the second page, but maybe it should be 10, same as on *Attendance History* page where arrow buttons are focusable.

   *Mark attendance* | *Edit attendance* 
   -- | --
   <img width="1461" height="1355" alt="Image" src="https://github.com/user-attachments/assets/fdc4cd37-e0d8-45cf-9bb4-b0c4483a51a3" /> | <img width="1406" height="1200" alt="Image" src="https://github.com/user-attachments/assets/6ba7f7c5-a258-4a3c-a81d-3ee47c81619d" />

2. Other accessibility issues include the 🔍 icon in the search field which is a ``span`` with no role, but with ``aria-label`` (throws a validation error), and insufficient contrast for absent learners in the count summary at the bottom of *Mark attendance* and *Edit attendance* pages.

   <img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/7e099a60-12a4-4fc0-8a1b-250fbf5011e4" />

   Since 🔍 is purely decorative icon, and the search field has its own ``aria-label``, let's just remove the ``aria-label`` from the icon's span.

3. In Chrome on both Windows and Ubuntu there are some weird and unnecessary horizontal scrollbars in all the cells with toggles in *Mark attendance* and *Edit attendance* pages, plus a vertical one in the bottom bar. Firefox on Windows displays only the latter, while Firefox on Ubuntu displays none (winner! 😂)

   *Firefox* | *Chrome* 
   -- | --
   <img width="1557" height="1066" alt="Image" src="https://github.com/user-attachments/assets/6dd9eb15-2a74-4715-ac7d-8dfb8442092d" /> | <img width="1420" height="1163" alt="Image" src="https://github.com/user-attachments/assets/c5827b5d-05e3-4c2a-8b0e-88c19bfba633" />

4. (low priority) Buttons in the modal are too close and focus outline overlaps.
   1 | 2 
   -- | --
   <img width="578" height="311" alt="Image" src="https://github.com/user-attachments/assets/c05ecfd2-991b-407f-9b51-918521bfebd2" /> | <img width="585" height="323" alt="Image" src="https://github.com/user-attachments/assets/1ec7c1e4-e096-42c3-a52c-858c02d06e29" />

**Comment by marcellamaki** [comment]
> the list was still not paginated

the pagination threshold for marking attendance is 50 learners, because we anticipate groups of 35-45 learners per VE mentor/coach, and wanted to be sure they could mark all of the learners on a single screen via scroll without having to navigate back and forth across pages. if we get feedback from them that more pagination is better, we can change it. for now, let's not worry about it.

> the search field which is a span with no role

does 
```
<form role="search">
  <input type="search">
</form>
```
make more sense, @radinamatic ? 


> insufficient contrast for absent learners 

let's just use the same text color for both for now

> some weird and unnecessary horizontal scrollbars in all the cells with toggles in Mark attendance and Edit attendance pages, plus a vertical one in the bottom bar

i do not know what is going on here but let's definitely fix it 😅 

>  Buttons in the modal are too close and focus outline overlaps

yes, this is a P1 but hopefully a very quick fix

**Comment by radinamatic** [comment]
```
<form role="search">
  <!-- search input -->
</form>
```
is definitely an option, but wouldn't it be easier to just remove the ``aria-label`` from the span where the icon is? Screen reader output it Ok, it's just triggering validation errors, as non-semantic element as ``span`` without a proper role is not allowed ``aria-label`` 

Since search input field is properly announced by the screen reader as it has its own ``aria-label`` , adding the ``aria-label``  to the 🔍 icon that precedes it is superfluous, imho

**Comment by marcellamaki** [comment]
> wouldn't it be easier to just remove the aria-label from the span where the icon is? 

ah, yes, i misunderstood the issues there -- that sounds better to me

**Comment by akolson** [comment]
@marcellamaki @radinamatic, just noting here that we don't have a span, if anything its managed by KeenUI's `UiIcon` so we will be removing the aria-label setting for this. Also worth noting that this change affects other parts of the code base that use the `PaginatedListContainer` component via the `FilterTextbox` component

**Comment by marcellamaki** [comment]
so, does this for example have the same underlying implementation as the facility users filtering, @akolson ? 

<img width="1499" height="744" alt="Image" src="https://github.com/user-attachments/assets/e4248bc4-8a6b-495d-b283-a8dfd67c7e33" />

**Comment by marcellamaki** [comment]
I think if it's the same implementation as in other places (and we're just noticing the a11y considerations), let's not make a wider fix right now @akolson, and target that to our next release. i can coordinate with @radinamatic  on that piece

**Comment by akolson** [comment]
> so, does this for example have the same underlying implementation as the facility users filtering, @akolson ?

Accessibility wise --yes, for any table that uses the `PaginatedListContainer`. I however don't think the above is using the search field block (within `PaginatedListContainer`) that this change affects. The only component using the search field block in the `PaginatedListContainer` is the `IndividualLearnerSelector` although it seems quite used, so it's probably best to hold off any changes for now!

**Comment by marcellamaki** [comment]
okay -- let's plan to have this in an upcoming release then, rather than here 👍 thanks for the extra details

**Comment by marcellamaki** [comment]
resolved those easily fixable in #14398 , the rest are out of scope
